### PR TITLE
Only set focus on FormField if it containsFocus in the DOM

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR adjusts FormField setFocus to check if the FormField contains DOM focus. If true, it sets focus to true and applies focus styling. If false, it does not apply focus styling since another DOM element has focus.

This resolves a bug where the TextInput in Select + Search was displaying focus on both the FormField and the TextInput. See "Background context" below for more details.

with fix
Screen Shot 2021-03-19 at 1 00 43 PM

without fix
Screen Shot 2021-03-19 at 1 01 07 PM

#### Where should the reviewer start?
Formfield.js
#### What testing has been done on this PR?
storyboook
#### How should this be manually tested?
When `select` is in `formfield` there should only be one focus
```
        <FormField>
          <Select
            size="medium"
            placeholder="Select single option"
            value={value}
            options={options}
            onChange={({ option }) => setValue(option)}
            onClose={() => setOptions(defaultOptions)}
            onSearch={text => {
              // The line below escapes regular expression special characters:
              // [ \ ^ $ . | ? * + ( )
              const escapedText = text.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&');

              // Create the regular expression with modified value which
              // handles escaping special characters. Without escaping special
              // characters, errors will appear in the console
              const exp = new RegExp(escapedText, 'i');
              setOptions(defaultOptions.filter(o => exp.test(o)));
            }}
          />
        </FormField>

```
#### Any background context you want to provide?

Select with Search applies focus onto the Search Input when the Select Drop opens. However, FormField is controlling its own focus internally and was not aware that the DOM focus had been placed on the search input in the select drop. This resulted in a focus indicator being placed on both the search input and the formfield (even though the Search Input was the active DOM element). By adding this check, we ensure that FormField only sets its focus to true when it also has focus in the DOM.

#### What are the relevant issues?
https://github.com/grommet/grommet-theme-hpe/issues/171
#### Do the grommet docs need to be updated?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Review by taysea — APPROVED** [review]
This LGTM.

**Comment by ShimiSun** [comment]
Can we please not update `yarn.lock` as part of this PR?

File: src/js/components/FormField/FormField.js [hunk 1]
```diff
@@ -8,7 +8,12 @@ import React, {
 import styled, { ThemeContext } from 'styled-components';
 import { defaultProps } from '../../default-props';
 
-import { focusStyle, parseMetricToNum } from '../../utils';
+import {
+  containsFocus,
+  useForwardedRef,
+  focusStyle,
+  parseMetricToNum,
+} from '../../utils';
```
Comment (ShimiSun) [inline src/js/components/FormField/FormField.js:16]: Please use explicit imports and alphabetized order. e.g.
```suggestion
import {
  containsFocus,
} from '../../utils/DOM';
....
```

**Comment by ShimiSun** [comment]
Also @taysea, can you please add details on why does this fix solves the problem. It will help with the context of the solution.

**Comment by taysea** [comment]
> Also @taysea, can you please add details on why does this fix solves the problem. It will help with the context of the solution.

Yes, I'll add to the PR description.

File: src/js/components/FormField/FormField.js [hunk 2]
```diff
@@ -8,7 +8,10 @@ import React, {
 import styled, { ThemeContext } from 'styled-components';
 import { defaultProps } from '../../default-props';
 
-import { focusStyle, parseMetricToNum } from '../../utils';
+import { containsFocus } from '../../utils/DOM';
+import { useForwardedRef } from '../../utils/refs';
```
Comment (taysea) [inline src/js/components/FormField/FormField.js:12]: Can we move this to the bottom? so the imports are alphabetized

**Review by ShimiSun — APPROVED** [review]

**Comment by ShimiSun** [comment]
Good stuff ladies!! ❤️ 

@leossantiago I thought you might e interested in seeing this fix 💟

**Pull request merged**